sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/RequiredServiceVersion.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.test.shared.extensions;
+
+import org.spockframework.runtime.extension.ExtensionAnnotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+@ExtensionAnnotation(RequiredServiceVersionExtension.class)
+public @interface RequiredServiceVersion {
+    Class<? extends Enum<?>> clazz();
+    String min();
+}
```

Comment:
This isn't the prettiest. These are boundaries:
- java annotation has quite restrictive list of types that can be used to declare it's values.
- We don't want to repeat this code and accompanying extension across modules. (to make it worse nio and batch would need same copy as blobs would have..)

I'm open to suggestions here.